src/Texture.cpp

```
@@ -48,8 +48,7 @@ Gosu::Texture::Texture(int width, int height, bool retro)
 
 Gosu::Texture::~Texture()
 {
-    ensure_current_context();
-
+    const auto lock = lock_gl_context();
     glDeleteTextures(1, &m_tex_name);
```

Should probably swallow exceptions here.